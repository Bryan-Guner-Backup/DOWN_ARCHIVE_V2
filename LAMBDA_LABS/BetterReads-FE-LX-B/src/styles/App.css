.bookContainer{
    display: flex;
}

.bookContainer .bookThumbStatus {
    display: flex;
    flex-direction: column;
}

.bookContainer .bookDetails{
    display: flex;
    flex-direction: column;   
}

.bookContainer .bookDetails .bookTitleFavIcon{
    display: flex;
    flex-direction: row;
}