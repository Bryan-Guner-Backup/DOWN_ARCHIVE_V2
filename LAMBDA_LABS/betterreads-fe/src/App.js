import React from 'react';
import { Router, Route, Switch } from 'react-router-dom';
import { PrivateRoute } from './utils/PrivateRoute';
// Components
import Library from './components/library/Library';
import SignIn from './components/authentication/SignIn';
import SignUp from './components/authentication/SignUp';
import PassReset from './components/authentication/PassReset';
import Success from './components/authentication/Success';
import Failure from './components/authentication/Failure';
import PageNotFound from './components/authentication/PageNotFound';
import Search from './components/search/Search';
import ShelfBook from './components/library/ShelfBook';
import Shelf from './components/library/Shelf';
import Shelves from './components/library/Shelves';
import OnboardingQuiz from './components/authentication/OnboardingQuiz';
import LandingPage from './components/Landing/Landing';
// History Util
import history from './utils/history';

const App = () => {
  return (
    <Router history={history}>
      <Switch>
        <PrivateRoute exact path='/home' component={Library} />
        <Route exact path='/' component={LandingPage} />
        <Route exact path='/signin' component={SignIn} />
        <Route exact path='/signup' component={SignUp} />
        <Route exact path='/reset' component={PassReset} />
        <Route exact path='/quiz' component={OnboardingQuiz} />
        <Route exact path='/success' component={Success} />
        <Route exact path='/failure' component={Failure} />
        <Route exact path='/pagenotfound' component={PageNotFound} />
        <PrivateRoute exact path='/search' component={Search} />
        <PrivateRoute exact path='/book/:id' component={ShelfBook} />
        <PrivateRoute exact path='/shelf/:shelf' component={Shelf} />
        <PrivateRoute exact path='/myshelves' component={Shelves} />
        <Route component={PageNotFound} />
      </Switch>
    </Router>
  );
};

export default App;
