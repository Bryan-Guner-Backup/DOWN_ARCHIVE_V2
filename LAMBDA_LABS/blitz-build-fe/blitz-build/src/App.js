//notes
import React, { useState, useEffect } from "react";
import { Switch } from "react-router";
import { BrowserRouter as Router, Route } from "react-router-dom";
// import Login from "./components/auth/Login";
// import Signup from "./components/auth/Signup";
// import PrivateRoute from "./components/auth/PrivateRoute";
import NavBar from "./components/NavBar";
import jwtDecode from "jwt-decode";
// import Layout from "./components/dashboard/Layout";
// import Dashboard from "./components/dashboard/index";
// import Dashboard from "./components/dashboard/Dashboard";

//task views
import Tasks from "./views/tasks/Tasks";
import ProjectTasks from "./views/tasks/ProjectTasks";

import Templates from "./components/templates/templates";
import IndividualTemplate from "./components/templates/IndividualTemplate";
import NinetyDayTemplate from "./components/templates/90DayTemplate";
import Projects from "./views/projects/Projects";
import IndividualProject from "./views/projects/IndividualProject";
import Logout from "./components/auth/Logout";
import Layout from "./layouts/Layout";
import Dashboard from "./components/dashboard/Dashboard";
import DelayLog from "./views/delayLog/DelayLog";
import OpenTemplateContext from "./contexts/OpenTemplateContext";

import Documents from "./components/documents/Documents";
//SWITCH INDEX TO DASHBOARD AFTER LC CHANGES HIS FILE NAME
import ActivityViewAll from "./components/activityFeed/ActivityViewAll";

import Landing from "./components/landing/pages/Landing";

//context
import UserContext from "./contexts/UserContext";
import TaskProvider from "./contexts/tasks/TaskProvider";
import DelayLogProvider from "./contexts/delayLog/DelayLogProvider";
import PathnameContext from "./contexts/PathnameContext";
import ProjectsProvider from "./contexts/projects/ProjectsProvider";
import EditModalContext from "./contexts/EditModalContext";
import SearchProvider from "./contexts/searching/searchTermProvider";
// import SearchTermContext from "./contexts/searching/searchTerm";
// import TemplateProvider from "./contexts/templates/TemplateProvider";
// import { axiosWithAuth } from "./utils/auth/axiosWithAuth";
import ComponentToPrint from "./components/documents/PrintDocument";
//AUTH0
import Auth from "./components/auth/auth";
import AuthNavBar from "./components/auth/authNavBar";
import Callback from "./components/auth/callback";
import ViewDocument from "./components/documents/ViewDocument";
import TemplatesProvider from "./contexts/templates/TemplateProvider";
import Uploader from "./components/documents/Uploader";
import MyCalendar from "./components/calendar/MyCalender";
import DocumentsProvider from "./contexts/documents/DocumentsProvider";
function App() {
  const [pathname, setPathname] = useState(window.location.pathname);
  const [openTemplate, setOpenTemplate] = useState(false);
  const [userInfo, setUserInfo] = useState([]);
  const [editModalOpen, setEditModalOpen] = useState(false);

  // getting userInfo from id_token in localStorage.
  useEffect(() => {
    if (localStorage.getItem("id_token")) {
      setUserInfo(jwtDecode(localStorage.getItem("id_token")));
    }
  }, []);

  //function that checks the path to see if it should return the marketing page
  const LandingPage = component => {
    if (pathname === "/") {
      return null;
    } else {
      return component;
    }
  };

  // console.log("userInfo", userInfo);
  console.log("pathname", pathname);
  const navLinks = [
    {
      text: "Home",
      path: "/dashboard",
      icon: "ion-ios-home"
    },
    {
      text: "Projects",
      path: "/projects",
      icon: "ion-ios-construct"
    },
    {
      text: "Tasks",
      path: "/tasks?filter=ACTIVE",
      icon: "ion-ios-notifications"
    },
    {
      text: "Documents",
      path: "/documents",
      icon: "ion-ios-document"
    },
    {
      text: "Templates",
      path: "/templates",
      icon: "ion-ios-menu"
    },
    {
      text: "Delay Log",
      path: "/delay-log",
      icon: "ion-ios-hourglass"
    },
    {
      text: "MyCalendar",
      path: "/mycalendar",
      icon: "ion-ios-calendar"
    },
    {
      text: "Log Out",
      path: "/log-out",
      icon: "ion-ios-cog"
    },
    {
      text: "Help",
      path: "/help",
      icon: "ion-ios-help-circle-outline"
    }
  ];

  return (
    <Router>
      <TemplatesProvider>
        <OpenTemplateContext.Provider value={{ openTemplate, setOpenTemplate }}>
          <ProjectsProvider>
            <DocumentsProvider>
              <TaskProvider>
                <DelayLogProvider>
                  <SearchProvider>
                    <EditModalContext.Provider
                      value={{ editModalOpen, setEditModalOpen }}
                    >
                      <PathnameContext.Provider
                        value={{ pathname, setPathname }}
                      >
                        <UserContext.Provider value={{ userInfo, setUserInfo }}>
                          {LandingPage(
                            <NavBar
                              setPathname={setPathname}
                              navLinks={navLinks}
                            />
                          )}
                          <Layout pathname={pathname}>
                            <Switch>
                              <Route exact path="/" component={Landing} />

                              <Route
                                exact
                                path="/navbar"
                                component={AuthNavBar}
                              />
                              <Route
                                exact
                                path="/callback"
                                component={Callback}
                              />
                              <Route
                                exact
                                path="/activity-feed"
                                component={ActivityViewAll}
                              />
                              <Route
                                exact
                                path="/login"
                                // component={Login}
                              />

                              {/* <Route exact path="/signup" component={Signup} /> */}
                              <Route exact path="/log-out" component={Logout} />
                              {/*   */}
                              <Route
                                exact
                                path="/dashboard"
                                component={Dashboard}
                              />
                              <Route exact path="/tasks" component={Tasks} />
                              <Route
                                exact
                                path="/projects"
                                component={Projects}
                              />
                              <Route
                                exact
                                path="/projects/:id"
                                component={IndividualProject}
                              />
                              <Route
                                exact
                                path="/projects/:id/tasks"
                                component={ProjectTasks}
                              />
                              <Route
                                exact
                                path="/templates"
                                component={Templates}
                              />
                              <Route
                                exact
                                path="/templates/:id"
                                component={IndividualTemplate}
                              />
                              <Route
                                exact
                                path="/90_Day"
                                component={NinetyDayTemplate}
                              />
                              <Route
                                exact
                                path="/delay-log"
                                component={DelayLog}
                              />
                              <Route
                                exact
                                path="/documents"
                                component={Documents}
                              />
                              <Route
                                exact
                                path="/documents/add"
                                component={Uploader}
                              />
                              <Route
                                exact
                                path="/documents/view/:file_name"
                                component={ViewDocument}
                              />
                              <Route
                                exact
                                path="/documents/print"
                                component={ComponentToPrint}
                              />
                              <Route
                                exact
                                path="/mycalendar"
                                component={MyCalendar}
                              />
                            </Switch>
                          </Layout>
                        </UserContext.Provider>
                      </PathnameContext.Provider>
                    </EditModalContext.Provider>
                  </SearchProvider>
                </DelayLogProvider>
              </TaskProvider>
            </DocumentsProvider>
          </ProjectsProvider>
        </OpenTemplateContext.Provider>
      </TemplatesProvider>
    </Router>
  );
}

export default App;
