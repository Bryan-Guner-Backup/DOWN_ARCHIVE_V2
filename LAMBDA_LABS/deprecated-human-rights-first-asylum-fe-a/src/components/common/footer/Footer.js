import React from 'react';
import { Menu } from 'antd';
import { StyledFooter } from './Footer-styling';

const Footer = () => {
  return (
    <>
      <StyledFooter>
        <Menu mode="horizontal">
          <Menu.Item>
            <a href="#" style={{ color: 'white' }}>
              Contact
            </a>
          </Menu.Item>
          <Menu.Item>
            <a href="#" style={{ color: 'white' }}>
              Privacy
            </a>
          </Menu.Item>
          <Menu.Item>
            <a href="#" style={{ color: 'white' }}>
              Terms
            </a>
          </Menu.Item>
        </Menu>
      </StyledFooter>
    </>
  );
};

export default Footer;
