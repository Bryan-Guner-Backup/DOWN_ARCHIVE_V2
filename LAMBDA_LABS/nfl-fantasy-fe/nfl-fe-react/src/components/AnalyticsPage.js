import React from "react";
import styled from "styled-components"; 
import MAE from "../styling/images/MAE.png";
import PercentCorrect from "../styling/images/PercentCorrect.png"; 
import LargeDivider from "../styling/LargeDivider.js"; 

const AnalyticsPage = () => {
    return (
        <div className="AnalyticsPage">
            <Heading className="Title">
            <h1>Methodology</h1> <LargeDivider />
            </Heading>
            <Div className="Intro"> 
            <Titles>Introduction 
                <Divider className="styling" />
                </Titles> 
            <P>NFL fantasy football is a game in which football fans take on the role of the coach or general manager of a pro football team. Partipants draft their initial teams, select players to play each week and trade players in order to compete weekly during a season against other teams. The winning teams are determined by the real-life statistics of the pro athletes. Goal of the Trade Analyzer
            The goal of the trade analyzer is to determine if a proposed trade of one or more players is a “good” or “bad” trade during any week of the 2019 football season. A good trade is defined as trading a player on your team for a player with higher expected fantasy football points. The goal, therefore, is to determine the expected fantasy football points of each pro football player for each week of the 2019 season. The expected points can then be compared between players. </P> 
            <Titles>The Points System <Divider className="styling" /></Titles> 
            <P>The trade analyzer uses the ESPN Standard League rules for determining fantasy football points. A team consists of 9 starters in 7 position. The 7 postions are quarterback(QB), running back(RB), wide receiver(WR), tight end(TE), kicker(K) and team defense(DF). A team’s defensive squad is considered one “player” in fantasy football. <br/> See <a href='https://nfl-trade.herokuapp.com'> nfl-trade.herokuapp.com</a> 
            {/* for the table that should go here (data for the table is the section below) */}
            <br/> 
            {/* QB/RB/WR/TEKickersDefense and Special Teams6 pts rush or receive TD5 pts 50+ yard FG made6 pts def/special teams TD6 pts returning kick/punt for TD4 pts 40-49 yard FG made2 pts interception6 pts player ret/rec fumble for TD3 pts FG made 39 yards or less2 pts fumble recovery4 pts passing TD2 pts rush/pass/rec 2 pt. conv2 pts blocked punt/PAT/FG2 pts rush/receive 2 pt. conv1 pt extra point made2 pts safety2 pts passing 2 pt. conversion-1 pt missed FG4 pts 1-6 points allowed1 pts 10 yards rush or receive3 pts 7-13 points allowed1 pt 25 yards passing1 pts 14-17 points allowed2 pts rush or receive TD of 40+ yds0 pts 18-27 points allowed2 pts passing TD of 40+ yds-1 pts 28-34 points allowed-2 pts interception-3 pts 35-45 points allowed-2 pts fumble lost-5 pts 46+ points allowed1 pt per sack */}
            </P> <br/>
            </Div>
            <Div className="Data">
            <Titles>Data <Divider3 className="styling" /> </Titles> 
            <P>The data used for modeling came from ArmchairAnalysis.com. The data consists of over 700+ data points over 28 different excel tables providing 20 years of detailed stats for every player in the NFL since the year 2000. The raw data was initially manipulated in Excel in order to link and extract the relevant data for all 2019 NFL players. The manipulated data formed the basis for the initial modeling data. From the initial modeling data the actual fantasy football points were calculated for each game of each season for every 2019 player. The training data for the models used all the data available for each player up through the 2018 season with the 2019 data used as the test data. For example, if a player begin his pro career in 2014, then 2014 - 2018 data was used for training data and 2019 data was used for test data.</P> <br/>
            </Div>
            <Div className="Modeling">
            <Titles >Modeling <Divider3 className="styling" /> </Titles> 
            <P>A total of four models were employed in predicting expected fantasy football points for 653 2019 NFL players. The data is essentially time-series data with players having anywhere from 0 to 20 years (20 seasons x 16 games = 320 data points) of data. <br />
            <br />
            1. Average - A baseline measure was established using an average for each 2019 player. The week 1 average was established using the average points per game for the players during the 2018 season (excluding rookies). Weeks 2 through 17 were calculated using the average of their actual performance from previous weeks (e.g. week 2 used week 1 actual performance, week 3 used the average of week 1 and 2 actuals, etc.). This would be considered the naive approach and one often used by the average fantasy football participant. <br/>
            <br />
            2 .XGBoost - An XGBoost model was used to predict rookie player performance in week 1 of the 2019 season. The 106 2019 rookies are difficult to predict as they have no professional experience. Using the rookie year stats for the 515 non-rookie players (excluding defense) as training data, week 1 expected fantasy points for the 2019 season were calculated with a validation MAE of 28.93 and R-squared of .49 and a test MAE of 30.75 and R-squared of .43. These were weak results, but a starting point was needed. Weeks 2 through 17 were calculated using the average of their actual performance from previous weeks - the same as the baseline (e.g. week 2 used week 1 actual performance, week 3 used the average of week 1 and 2 actuals, etc.). <br />
            <br />
            3. ARIMA - An ARIMA model was initially used to predict veteran player performance for each week in the 2019 season. Only veteran players with over 3 years of experience and at least 50 points during the 3 years were run using ARIMA as the model needs about 50 data points (3 seasons x 16 games = 48 data points) to perform well. About 246 veteran players were modeled using ARIMA and while the results were very good (see the graphs) and beat the baseline, the neural network model showed better results. <br />
            <br />
            4. Recurrent Neural Network, Long Short-Term Memory (RNN-LSTM) - A RNN-LSTM model was used to predict veteran player performance for each week in the 2019 season. Only veteran players with over 3 years of experience and at least 50 points during the 3 years were run using RNN-LSTM as the model needs as much data as possible and 3 years seemed to be a good balance. About 246 veteran players were modeled using RNN-LSTM and the results compared to the ARIMA model. The RNN-LSTM model outperformed both the baseline average and the ARIMA models and was the model used in the final app.
            </P> <br />
            </Div>  
            <Div className="Notes">
            <Titles >Notes: <Divider3 className="styling" /></Titles> 
            <P>
            1. Final Results - The final results combine the rookie results(XGBoost+Average - 106 players), the veterans with 3+ years(RNN-LSTM - 278 players including defense) and the remaining veterans with more than 3 years experience(Average - 269 players). This is referred to as the “NN” model in the Evaluate tab and the “ARIMA” is the same combination of results only using ARIMA for the veterans with 3+ years. <br />
            <br />
            2. Weekly Predictions - The models each calculate predictions for each of the remaining weeks of the season. Each remaining week is then summed to arrive at the players expected fantasy football points. For example, in week 1 the expected fantasy football points are the sum of all 16 game predictions (the entire season), in week 2 the points are the sum of the 15 remaining games, etc. <br />
            <br />
            3. Bye Weeks - There are actually 17 weeks in a football season and each team has a random bye week during the season. On bye weeks, each player’s prediction from the week before is simply carried forward. <br />
            <br /> 
            4. Injuries - Injuries play a huge role in fantasy football. When a player is injured his points essentially fall to zero. There is no reliable way to forecast if or when a player will get injured. While the raw data does provide some details on injuries, it still does not provide good information on when a player will return from an injury. In all the models, when a player goes on injured reserve status, he is assumed to be out for the season. <br /> 
            <Titles2>An Evaluation of the Model Predictions <Divider2 className="styling" /></Titles2> 
            The Recurrent Neural Network - LSTM Model(NN) outperforms both the baseline Average and ARIMA models. While both the NN and ARIMA models have lower MAEs they all start to converge about halfway through the season as all models incorporate the current season actual points and the number of future predictions are reduced thereby increasing accuracy in all the models. The real accuracy of the models are shown in the Percent of Correct Predictions graph. For each week, the predicted outcome of a “good” or “bad” trade for each player is calculated against every other player and compared with the actual outcome. The NN clearly makes better predictions over the other models with an overall average correct percentage of 85.59% for the 2019 season compared with 84.72% for an ARIMA model and 83.17% for the baseline average model. </P> <br/>
            </Div>
        <div className="graphs">
            <div className="image-1">
                <img src={MAE} alt="Mean Average Error for Each Model" height="500px" />
            </div>
            <div className="image-2">
                <img src={PercentCorrect} alt="Percent of Correct Predictions for Each Model" height="500px" />
            </div>
        </div>
        </div>
    )
}

export default AnalyticsPage; 

const Heading = styled.h3`
background: #ffa630;
font-size: 3.5rem;
width: 75%;
margin: auto
`

const Titles = styled.div`
padding: 1.5rem; 
font-size: 2rem;  
`

const Titles2 = styled.div`
padding: 1.5rem;
font-size: 2rem;  
text-align: center;
`

const Div = styled.div`
width: 65%; 
margin: auto;
`

const Divider = styled.div`
border: 1px solid black; 
width: 50%;
margin: auto;
`

const Divider2 = styled.div`
border: 1px solid black; 
width: 95%;
margin: auto;
`

const Divider3 = styled.div`
border: 1px solid black; 
width: 30%;
margin: auto;
`

const P = styled.p`
text-align: left; 
`