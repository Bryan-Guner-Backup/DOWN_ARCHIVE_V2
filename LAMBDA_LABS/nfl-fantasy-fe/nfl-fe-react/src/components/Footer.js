import React from 'react';

import styled from "styled-components"; 

const Footer = () => {
    return (
    <Div className="Footer">
        <Border className="styling">
        <div className="Link-Area">
            <CopyRightDiv className="Copyright-Info">
            <h5>Copyright Â© 2020. All Rights Reserved, built by Students at <A href="https://lambdaschool.com/">Lambda School </A> </h5>
            </CopyRightDiv>
        </div>
        </Border> 
    </Div>
    ); 
};

export default Footer; 

const Div = styled.div`
width: 65%;
margin: auto;
padding: 2rem;
justify-content: center;
`

const Border = styled.div`
border-top: 1.25px solid #001021; 
`

const CopyRightDiv = styled.div`
color: 	#696969;
text-align: center; 
`

const A = styled.a`
text-decoration: none;
color: 	#696969; 
cursor: pointer; 
&:hover {
    color: #001021;
    }
`