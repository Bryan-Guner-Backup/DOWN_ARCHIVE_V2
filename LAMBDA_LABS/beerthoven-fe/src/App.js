import React from "react";
import { Route, Switch } from "react-router-dom";
import "./App.css";
import { BrowserRouter as Router, Link } from "react-router-dom";
import { Layout, Menu } from "antd";
import { Security, SecureRoute, LoginCallback } from "@okta/okta-react";
import oktaConfig from "./okta-config";
import OktaApolloProvider from "./components/utils/OktaApolloProvider";
import LogoutButton from "./components/LogoutButton";
import Dashboard from "./components/pages/Dashboard";
import Login from "./components/pages/Login";
import People from "./components/pages/People";
import Event from "./components/pages/Event";
import Venue from "./components/pages/Venue";
import User from "./components/pages/User";
const { Sider, Content } = Layout;

function App() {
  return (
    <div className="App" data-testid="app">
      <Router>
        <Security {...oktaConfig}>
          <OktaApolloProvider>
            <Layout>
              <Sider
                style={{
                  overflow: "auto",
                  height: "100vh",
                  position: "fixed",
                  left: 0,
                }}
              >
                <Menu theme="dark">
                  <Menu.ItemGroup key="h1" title="Manage">
                    <Menu.Item key="1">
                      <Link to="/">Dashboard</Link>
                    </Menu.Item>
                    <Menu.Item key="2">
                      <Link to="/people">People</Link>
                    </Menu.Item>
                    <Menu.Item key="3">
                      <Link to="/event">Events</Link>
                    </Menu.Item>
                    <Menu.Item key="4">
                      <Link to="/venue">Venues</Link>
                    </Menu.Item>
                    <Menu.Item key="5">
                      <Link to="/user">Users</Link>
                    </Menu.Item>
                  </Menu.ItemGroup>
                </Menu>
                <div
                  style={{
                    paddingTop: "30px",
                    display: "grid",
                    placeItems: "center",
                  }}
                >
                  <LogoutButton />
                </div>
              </Sider>
              <Layout
                style={{
                  marginLeft: 200,
                  height: "100vh",
                }}
              >
                <Content style={{ margin: "24px 16px 0", overflow: "initial" }}>
                  <Switch>
                    <SecureRoute path="/login" component={Login} />
                    <SecureRoute path="/people" component={People} />
                    <SecureRoute path="/event" component={Event} />
                    <SecureRoute path="/venue" component={Venue} />
                    <SecureRoute path="/user" component={User} />
                    <Route
                      path={oktaConfig.callbackPath}
                      component={LoginCallback}
                    />
                    <SecureRoute exact path="/" component={Dashboard} />
                  </Switch>
                </Content>
              </Layout>
            </Layout>
          </OktaApolloProvider>
        </Security>
      </Router>
    </div>
  );
}

export default App;
