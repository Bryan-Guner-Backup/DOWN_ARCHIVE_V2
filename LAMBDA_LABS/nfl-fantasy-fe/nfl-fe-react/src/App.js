import React from "react";
import AppRouter from './components/AppRouter.js';
import './App.css';
import ReactGA from 'react-ga';
import { createBrowserHistory } from 'history';
import Footer from "./components/Footer.js";
import Nav from "./components/Nav.js";
import styled from "styled-components"; 


// Google Analytics for the whole app
    ReactGA.initialize('UA-145773395-3');
    const history = createBrowserHistory();
    // Initialize google analytics page view tracking
    history.listen(location => {
        ReactGA.set({ page: location.pathname }); // Update the user's current page
        ReactGA.pageview(location.pathname); // Record a pageview for the given page
    });
    history.listen(location => ReactGA.pageview(location.pathname));


function App() {
    return (
        <div className="App">
            <div className="Nav-Area">
                <Nav />
            </div>

            <MainDiv className="Main-Area">
            <h2> Fantasy Football Trade Analyzer </h2>
            <AppRouter/>
            </MainDiv>

            <div className="Foot-Area">
            </div>
                <Footer />
            </div>
    );
};


export default App;

const MainDiv = styled.div`
width: 85%;
margin: 0 auto; 
padding: 3rem; 
`