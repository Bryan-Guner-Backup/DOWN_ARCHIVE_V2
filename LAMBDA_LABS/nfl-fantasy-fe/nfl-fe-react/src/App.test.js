import React from 'react';
import * as rtl from "@testing-library/react";

import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import App from './App';
import Footer from './components/Footer';
import AnalyticsPage from './components/AnalyticsPage';

const mockStore = configureStore([]);

describe('My Connected React-Redux Component', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            myState: 'sample text',
        });

        store.dispatch = jest.fn();

        component = renderer.create(
            <Provider store={store}>
                <App />
            </Provider>
        );
    });

    it('renders without crashing', () => {
        rtl.render(
            <Provider store={store}>
                <App />
            </Provider>);
    })

    it('should render with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
   
    it('should dispatch an action on button click', () => {
        renderer.act(() => {
            component.root.findByType('button').props.onClick();
        });
    });

    it('renders Copyright on Footer', () => {
        const {getByText} = rtl.render(<Footer />);
        const CopyRight = getByText(/Copyright/i);
        expect(CopyRight).toBeInTheDocument();
    })

    it('renders Analytics Page', () => {
        const {getByText} = rtl.render(<AnalyticsPage />);
        const Data = getByText(/Methodology/i);
        expect(Data).toBeInTheDocument();
    })
  });
