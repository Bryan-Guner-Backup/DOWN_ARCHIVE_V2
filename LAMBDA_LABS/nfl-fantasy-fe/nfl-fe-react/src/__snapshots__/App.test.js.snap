// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="App"
>
  <div
    className="Nav-Area"
  >
    <div
      className="sc-fzpjYC iJVkNb nav"
    >
      <div
        className="Logo-Area"
      >
        <img
          alt="tacklemytrade.com"
          height="150px"
          src="tacklemytradelogo.jpg"
        />
      </div>
      <div
        className="sc-fznxsB eyWaoH Link-Area"
      >
        <a
          className="sc-fznJRM dMEeeu link"
          href="/"
        >
           Home 
        </a>
        <a
          className="sc-fznJRM dMEeeu link"
          href="/about-us"
        >
           About Us
        </a>
        <a
          className="sc-fznJRM dMEeeu link"
          href="/howtoguide"
        >
           How To Play 
        </a>
        <a
          className="sc-fznJRM dMEeeu link"
          href="/analytics"
        >
           Analytics 
        </a>
      </div>
      <div
        className="sc-fznxsB eyWaoH Right-Link-Area"
      >
        <a
          className="sc-fznJRM dMEeeu link"
          href="/register"
        >
           Register 
        </a>
        <a
          className="sc-fznJRM dMEeeu link"
          href="/login"
        >
           Login 
        </a>
      </div>
    </div>
  </div>
  <div
    className="sc-fznWqX kJmqqs Main-Area"
  >
    <h2>
       Fantasy Football Trade Analyzer 
    </h2>
    <div>
      <form
        className="ui form"
        onSubmit={[Function]}
      >
        <h3
          className="sc-AxjAm iLOmxz"
        >
           Team 1 
        </h3>
        <div
          className="field"
        >
          <div
            className="ui input"
          >
            <input
              aria-describedby={null}
              name="player0_name"
              onChange={[Function]}
              placeholder="1st player"
              type="text"
              value=""
            />
          </div>
        </div>
        <h3
          className="sc-AxjAm iLOmxz"
        >
           Team 2 
        </h3>
        <div
          className="field"
        >
          <div
            className="ui input"
          >
            <input
              aria-describedby={null}
              name="player1_name"
              onChange={[Function]}
              placeholder="2nd player"
              type="text"
              value=""
            />
          </div>
        </div>
        <h3
          className="sc-AxjAm iLOmxz"
        >
           Week 
        </h3>
        <div
          className="field"
        >
          <div
            className="ui input"
          >
            <input
              aria-describedby={null}
              name="week"
              onChange={[Function]}
              placeholder="week number"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          className="sc-AxirZ hCOwQG ButtonDiv"
        >
          <div
            className="field"
          >
            <button
              aria-describedby={null}
              className="ui button"
              onClick={[Function]}
              type="submit"
            >
              Submit
            </button>
          </div>
        </div>
      </form>
    </div>
    <div />
  </div>
  <div
    className="Foot-Area"
  />
  <div
    className="sc-fzqNJr iavbLC Footer"
  >
    <div
      className="sc-fzoyAV hFgxSs styling"
    >
      <div
        className="Link-Area"
      >
        <div
          className="sc-fzoLag jZJZET Copyright-Info"
        >
          <h5>
            Copyright Â© 2020. All Rights Reserved, built by Students at 
            <a
              className="sc-fzoXzr fYeqnm"
              href="https://lambdaschool.com/"
            >
              Lambda School 
            </a>
             
          </h5>
        </div>
      </div>
    </div>
  </div>
</div>
`;
